source/extensions/transport_sockets/alts/tsi_socket.cc

```diff
@@ -12,7 +12,7 @@ namespace Alts {
 namespace {
 // Include 4 bytes frame message type and 16 bytes tag length.
 // It is consistent with gRPC ALTS zero copy frame protector implementation.
-static constexpr int FrameOverheadSize = 0;
+static constexpr int FrameOverheadSize = 20;
```

Comment:
The use of max_unprotected_frame_size_ and setMaxUnprotectedFrameSize seems to make testing for this overhead constant difficult.  Or at least that's the impression I get from no tests needing update when changing this overhead value.  max_unprotected_frame_size_ is read in a single location in repeatProtectAndWrite, consider changing that to "actual_frame_size_to_use_ - FrameOverheadSize"

Reply:
We can not use a hard-coded `FrameOverheadSize` within the code because in the test we used fake frame protector which has a different frame overhead size (i.e., 4 bytes). The code is updated. PTAL.